newsroom/search/service.py

```diff
@@ -866,3 +825,61 @@ def get_matching_topics_for_item(self, topics, users, companies, query):
             raise
 
         return topic_matches
+
+    def get_topic_query(self, topic, user, company, section_filters, query=None):
```

Comment:
I've done something similar with my work on scheduled notifications. And as this is in the typescript branch, we will have merge conflicts. In this case, would it be better to base/merge this into the `develop` branch first then cherry-pick into typescript branch.

This way I can rebase from latest develop merging your changes with mine

Reply:
ok np